File: Roman to Integer/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        symbolToValueMap = {
        'I': 1, 'V': 5, 'X': 10, 'L':50,
        'C': 100, 'D': 500, 'M': 1000
        }
        
        number = 0
        for index, _val in enumerate(s):
            if (index + 1 < len(s)) and (symbolToValueMap[s[index]] < symbolToValueMap[s[index + 1]]):
                number -= symbolToValueMap[s[index]]
            else: 
                number += symbolToValueMap[s[index]]
        return number


classInstance = Solution()
print(classInstance.romanToInt('MCMXCIV'))

File: Roman to Integer/roman_to_integer.js
/**
 * @param {string} s
 * @return {number}
 */
 var romanToInt = function(s) {
    let symbolToValueMap = {
        'I': 1, 'V': 5, 'X': 10, 'L':50,
        'C': 100, 'D': 500, 'M': 1000
    }

    let number = 0
    s = s.split('')
    while(s.length) {
        const currentSymbol = s.shift()
        if(symbolToValueMap[currentSymbol] < symbolToValueMap[s[0]])
            number -= symbolToValueMap[currentSymbol]
        else
            number += symbolToValueMap[currentSymbol]
    }

    return number
};

console.log(romanToInt('MCMXCIV'))